### PR TITLE
Refactor css custom props

### PR DESCRIPTION
**Pull request by markacianfrani**
## Brief Description

* Removes all of the unused Astro 5 custom properties
* Converts all CSS custom props to dash-case for better compatibility with the lit repo.
* Component specific properties have been moved back to their respective component styles. 
* Some properties were removed entirely--for example cursor since it's not something that would really ever need to be changed. 

## JIRA Link

[ASTRO-1856](https://rocketcom.atlassian.net/browse/ASTRO-1856)
[ASTRO-1891](https://rocketcom.atlassian.net/browse/ASTRO-1891)

## Motivation and Context

Preparation for design token effort. Puts us in more in sync with design. All values were pulled programmatically from Figma using Style Dictionary as a POC of what the future design -> dev pipeline might look like.

## Issues and Limitations

* VRT fail in a number of places due to incorrect values on our end. These include: button secondary outline, table hover row, progress background color, notification icon color. 
* Some components aren't finished in Figma so there could be future discrepancies. 

## Types of changes

-   [x] Bug fix
-   [x] New feature
-   [x] Breaking change

## Checklist

-   [ ] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [ ] I have added tests to cover my changes.

### CODE REVIEW

File: src/components/rux-radio/rux-radio.scss [hunk 1]
```diff
@@ -7,42 +7,30 @@
...
-    /**
-  * @prop --controlSelectedOutlineBorderColor: Selected radio border color
-  */
-    --controlSelectedOutlineBorderColor: var(--primary);
-
-    /** @prop --controlSelectedOutlineBackgroundColor: Selected radio outline background color */
-    /** @prop --controlSelectedBorderColor: Selected radio border color */
+
+    /** @prop --radio-hover-border-color: Radio border color on hover*/
+    --radio-hover-border-color: var(--hover);
```
Comment (NicDominguez) [inline src/components/rux-radio/rux-radio.scss:12]: awesome! thanks for defining the new names for radio

File: src/components/rux-segmented-button/rux-segmented-button.scss [hunk 2]
```diff
@@ -1,53 +1,46 @@
...
+    /** @prop --segmented-button-selected-hover-text-color: Segmented button selected hover text color*/
+    --segmented-button-selected-hover-text-color: var(--default-text);
 
-    /**
-    * @prop --segmentedButtonSelectedHoverBorderColor: Segmented button selected hover border color
-    */
+    /** @prop --segmented-button-selected-hover-border-color: Segmented button selected hover border color*/
+    --segmented-button-selected-hover-border-color: var(--primary);
+
+    /** @prop --segmented-button-border-radius */
```
Comment (NicDominguez) [inline src/components/rux-segmented-button/rux-segmented-button.scss:42]: Did you want to add a description to this prop as the table looks strange with a blank slot.

File: src/global/_variables.scss [hunk 3]
```diff
@@ -1,767 +1,168 @@
...
+    --font-size: 1rem; /* 16px */
+    --font-size-xl: 1.125rem;
+    --font-size-lg: 1rem;
+    --font-size-md: 0.875rem;
+    --font-size-sm: 0.75rem;
+    --font-size-xs: 0.65rem;
+
+    --font-family-mono: 'Roboto Mono', monospace;
+
+    --color-critical: var(--color-status-critical-500, rgb(255, 42, 4));
```
Comment (NicDominguez) [inline src/global/_variables.scss:152]: Not a big deal, but do we want the fallback values to also be hex just to match with the props?

**Review by NicDominguez — APPROVED** [review]
Excellent job! This must have been really tedious so thank you for putting in all the work.

File: src/components/rux-button/rux-button.scss [hunk 4]
```diff
@@ -1,65 +1,64 @@
...
-     * @prop --buttonSecondaryBackgroundColor: Button secondary background color
+     * @prop --button-secondary-background-color: Button secondary background color
      */
+    --button-secondary-background-color: transparent;
 
     /**
-     * @prop --buttonSecondaryBorderColor: Button secondary border color
+     * @prop --button-secondary-border-color: Button secondary border color
      */
+    --button-secondary-border-color: var(--primary, rgb(0, 90, 143));
```
Comment (micahjones13) [inline src/components/rux-button/rux-button.scss:51]: var(--primary) is #4dacff which in turn is rgb(77, 152, 255) but the fallback provided here doesn't match up.

**Review by micahjones13 — APPROVED** [review]
This is awesome! Nice work

File: src/global/_variables.scss (hunk 3 above)
Comment (markacianfrani) [reply src/global/_variables.scss:152]: I was thinking about that. Fallbacks are sort of just haphazardly included right now. Some of them are even wrong. I was thinking we'd do another pass that'll focus specifically on fallback values which could be done programmatically as separate sass variables or something.

Comment (markacianfrani) [reply src/global/_variables.scss:152]: created https://rocketcom.atlassian.net/browse/ASTRO-1927 to address this

File: src/components/rux-segmented-button/rux-segmented-button.scss (hunk 2 above)
Comment (markacianfrani) [reply src/components/rux-segmented-button/rux-segmented-button.scss:42]: bless you for catching that

**Pull request merged**